### DISCUSSION TITLE
Is your Blocking Queue... Blocking? - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#308] [Announcements]
# Is your Blocking Queue... Blocking? - Gunnar Morling

Java’s BlockingQueue hierarchy is widely used for coordinating work between different producer and consumer threads.
When set up with a maximum capacity (i.e. a bounded queue), no more elements can be added by producers to the queue once it is full, until a consumer has taken at least one element.
For scenarios where new work may arrive more quickly than it can be consumed, this applies means of back-pressure,
ensuring the application doesn’t run out of memory eventually, while enqueuing more and more work items.

https://www.morling.dev/blog/is-your-blocking-queue-blocking/

**Comment by swiatek25** [comment]
Great stuff!

**Reply by gunnarmorling** [reply]
Thank you!

**Comment by crucoba** [comment]
I've learned a lot. Thank you.

**Reply by gunnarmorling** [reply]
Awesome, that's great to hear!
